carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -280,8 +280,19 @@ std::vector<lanelet::ConstLanelet> CARMAWorldModel::getLaneletsBetween(double st
     prioritized_lanelets.pop();
     output.push_back(pair.lanelet_);
   }
+  
+  //Sort lanelets according to shortest path
+  std::vector<lanelet::ConstLanelet> sorted_output;
+  for(auto llt : route_->shortestPath()){
+    for(int i=0; i < output.size();i++){
+      if(llt.id() == output[i].id()){
+        sorted_output.push_back(llt);
+        break;
+      }
+    }
+  }
```

Comment:
This loop should only run if shortest_path_only is set to true. There is no way to order the non-shortest route lanelets in the alternative case.